### ISSUE TITLE
rest_framework_gis support doesn't work for GeometryFilters

### ISSUE THREAD

**Issue by AndrewGuenther** [tfranzel/drf-spectacular#814]
**Describe the bug**
When using `GeometryFilter`, the rest_framework_gis extension can't resolve the proper field type

**To Reproduce**
Use `GeometryFilter` inside of a `FilterSet`

**Expected behavior**
I'd expect that the filter field is recognized as a `GeometryField` and a schema is generated as such.

**Comment by AndrewGuenther** [comment]
I file this as a bug rather than a feature only because support for this is 99% of the way there.

Schema generation makes it all the way to this line: https://github.com/tfranzel/drf-spectacular/blob/master/drf_spectacular/contrib/rest_framework_gis.py#L212

And then fails because `parent` is `None`. I don't know if there's a "correct" way to identify that `target` is part of a filter rather than a serializer, but naively you can just assume if `parent` is `None` that you're dealing with a generic `GeometryField` as the default.

Which also raises the question: Since `GeometryField` is a superset of all geometry field types, should that be the default type returned in case of an exception instead of `object`?

**Comment by tfranzel** [comment]
I have a fix for this pending. It was not quite as simple as fixing that line though. Once at that location, the model information is lost and so this has has to be caught earlier but I digress.

The real problem is that their chosen [query format](https://github.com/openwisp/django-rest-framework-gis#geometryfilter) is not supported by OpenAPI 3.0.3. 

```text
/region/?contains_geom={ "type": "Point", "coordinates": [ -123.26436996459961, 44.564178042345375 ] }
AKA
/region/?contains_geom=%7B%22type%22%3A%20%22Point%22%2C%20%22coordinates%22%3A%20%5B-122.49034881591797%2C%2037.76949349270407%20%5D%7D
``` 

Those are the supported shapes (object column): https://spec.openapis.org/oas/v3.0.3#style-examples

I can see no way to tell OpenAPI that this is a urlencoded JSON string. If we simply advertise it as a string, all structure would be lost. :disappointed:

**Comment by AndrewGuenther** [comment]
So, I think `string` may actually be the correct type here anyway. `GeometryFilter` query parameters don't just accept GeoJSON, they also accept WKT, HEXEWKB, and WKB. The only way you can really describe all of those as a superset would be `string`...

https://github.com/openwisp/django-rest-framework-gis/blob/master/README.rst?plain=1#L462-L465

**Comment by tfranzel** [comment]
good point. It's just dissapointing since we have the schema ready. A `oneOf` with string and object would be viable. In any case I defaulted to string for now. This can be improved at a later time.

I need to do a hotfix release because of #818, which includes fixes for this and #813. Should do for now. Feedback welcome though. We can pick this up again, once you had some play time with it.

**Comment by jokiefer** [comment]
i found this issue, cause my schema does not provide the specific geojson type as a `format` hint or `oneOf` reference.

I struggled also arround by adding this hints. The only possible way i found was to override the `get_schema_operation_parameters` from the original [django_filters](https://github.com/carltongibson/django-filter/blob/f38dcdd32d20095cb823cbd5e85946510f146e29/django_filters/rest_framework/backends.py#L132) package, cause i need field access to analyze which filter type it is. 

I also test to bring up an extension:

```python
class GeoJsonFilterFix(OpenApiFilterExtension):
    target_class = QueryParameterValidationFilter
    match_subclasses = True

    def get_schema_operation_parameters(self, auto_schema: AutoSchema, *args, **kwargs) -> List[dict]:
        schema = self.target.get_schema_operation_parameters(auto_schema.view)
        print("schema", schema)

        return []
```

But i  need to reproduce the original `get_schema_operation_parameters` function by getting the field class to hook in by any `GeometryFilter` fiter field class.

Is there any other possibility to extend schema for `GeometryFilter` without reproducing that basic code of the django_filters package?


What i need is something like:

```python
{
                'name': field_name,
                'required': field.extra['required'],
                'in': 'query',
                'description': field.help_text if field.help_text is not None else field.label if field.label is not None else field_name,
                'schema': {
                    'type': 'string',
                    'format': 'geojson'
                },
            }

```
